src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -308,18 +310,32 @@ RCLConsensus::Adaptor::onClose(
     }
 
     // Add pseudo-transactions to the set
-    if ((app_.config().standalone() || (proposing && !wrongLCL)) &&
-        ((prevLedger->info().seq % 256) == 0))
+    if (app_.config().standalone() || (proposing && !wrongLCL))
     {
-        // previous ledger was flag ledger, add pseudo-transactions
-        auto const validations = app_.getValidations().getTrustedForLedger(
-            prevLedger->info().parentHash);
-
-        if (validations.size() >= app_.validators().quorum())
+        auto seq = prevLedger->info().seq + 1;
+        if (((seq - 1) % 256) == 0)
         {
-            feeVote_->doVoting(prevLedger, validations, initialSet);
-            app_.getAmendmentTable().doVoting(
-                prevLedger, validations, initialSet);
+            // previous ledger was flag ledger, add fee and amendment
+            // pseudo-transactions
+            auto validations = app_.getValidations().getTrustedForLedger(
+                prevLedger->info().parentHash);
+            filterValsWithnUnl(validations, app_.validators().getnUnlNodeIDs());
```

Comment:
this function is called even when the amendment is not enabled?

Reply:
Same as the one below. Will discuss there.